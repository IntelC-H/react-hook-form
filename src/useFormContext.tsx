import * as React from 'react';
import { FormContextValues, FormProps } from './types';

export const FormGlobalContext = React.createContext<FormContextValues>(
  {} as FormContextValues,
);

export const useFormContext = () =>
  React.useContext<FormContextValues>(FormGlobalContext);

export function FormContext(props: FormProps) {
  const { children, ...rest } = props;
  return (
    <FormGlobalContext.Provider
      value={
        {
          ...rest,
        } as FormContextValues
      }
    >{children}</FormGlobalContext.Provider>
    
  );
}
