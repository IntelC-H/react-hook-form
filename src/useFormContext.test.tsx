import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { FormContext, useFormContext } from './useFormContext';
import { mount } from 'enzyme';

describe('FormContext', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      // @ts-ignore
      <FormContext>
        <div>child</div>
      </FormContext>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should have access to all methods with useFormContext', () => {
    const test = jest.fn();
    const Child = () => {
      // @ts-ignore
      const { test } = useFormContext();
      test();
      return null;
    };
    const TestHook = () => {
      return (
        // @ts-ignore
        <FormContext test={test}>
          <Child />
        </FormContext>
      );
    };

    mount(<TestHook />);

    expect(test).toBeCalled();
  });
});
