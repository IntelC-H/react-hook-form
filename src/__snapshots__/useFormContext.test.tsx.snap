// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContext should render correctly 1`] = `
<div>
  child
</div>
`;
